style_guide.md

```diff
@@ -0,0 +1,290 @@
+# Jsonnet Style Guide
+This style guide documents how we build schemas consistently with Jsonnet. For more general principles and styling rules, we follow the [Databricks Style Guide](https://github.com/databricks/jsonnet-style-guide#databricks-jsonnet-guide).
+## Folder structure
+- The following structure is used:
+
+    ```
+    england-wales/
+       lib/
+          common_rules.libsonnet
+          placeholders.libsonnet
+       <questionnaire>/
+          <questionniare>.jsonnet
+          lib/
+             rules.libsonnet
+          blocks/
+             <section-one>/
+                <block-one>.jsonnet
+                <block-two>.jsonnet
+             <section-two>/
+    northern-ireland/   
+    ```
+- Jsonnet source files are separated into two top-level folders - `england-wales` and `northern-ireland` 
+  - The file extension should be `.libsonnet`
+  - Each file is grouped by function, currently `common_rules` and `placeholders`
+- Each questionnaire has it's own folder, within this:
+  - A top-level questionnaire definition Jsonnet file 
+  - A `lib/` folder holds common Jsonnet used across multiple blocks
+    - The file extension should be `.libsonnet`
+    - Each file here can extend common Jsonnet from the questionnaire level lib folder
+    - `rules.libsonnet` extends the top level `lib/common_rules.libsonnet`
+  - A `blocks` folder:
+    - Subdivided into section folders
+    - Within each section folder a Jsonnet file per block
+  
+- We currently have the following questionnaires:
+
+    - `ccs`
+    - `communal-establishment`
+    - `household`
+    - `individual`
+
+- Blocks from the `individual` questionnaire are also used in the `household` questionnaire.
+
+## Global variables
+
+- Global variables are variables known at questionnaire launch. They don't change during the completion of a questionnaire.
+- Top-level arguments (`--tla-str`) should be used to pass variables that are used in 5 blocks or less.
+- External variables (`--ext-str`) should be used to pass variables that are used in more than 5 blocks.
+  - External variables are used for `census_date` and `region_code`.
+  - Example of an external variable in `answers`:
+    
+        ```
+        answers: [
+          {
+            id: 'date-of-birth-answer',
+            mandatory: true,
+            type: 'Date',
+            minimum: {
+              value: std.extVar('census_date'),
+              offset_by: {
+                years: -115,
+                months: -2,
+                days: -20,
+              },
+            },
+            maximum: {
+              value: 'now',
+            },
+          },
+        ],
+        ```
```

Comment:
There is something wrong with the formatting here as it isn't rendering as expected in the Markdown.

Reply:
Looks like it was spacing problem, it's fixed now👍